external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/CachedKafkaConsumer.scala

```diff
@@ -383,19 +362,16 @@ private[kafka010] object CachedKafkaConsumer extends Logging {
     // If this is reattempt at running the task, then invalidate cache and start with
     // a new consumer
     if (TaskContext.get != null && TaskContext.get.attemptNumber >= 1) {
-      removeKafkaConsumer(topic, partition, kafkaParams)
-      val consumer = new CachedKafkaConsumer(topicPartition, kafkaParams)
-      consumer.inuse = true
-      cache.put(key, consumer)
-      consumer
-    } else {
-      if (!cache.containsKey(key)) {
-        cache.put(key, new CachedKafkaConsumer(topicPartition, kafkaParams))
-      }
-      val consumer = cache.get(key)
-      consumer.inuse = true
-      consumer
+      cache.invalidate(key)
     }
+
+    val consumer = cache.get(key, new Callable[CachedKafkaConsumer] {
```

Comment:
I'm probably wrong about this, but does Scala let you write a simple lambda here? because it's an interface with just one method. Might be clearer if so.